uxarray/core/dataarray.py

```diff
@@ -249,6 +250,65 @@ def to_dataset(self) -> UxDataset:
         xrds = super().to_dataset()
         return uxarray.core.dataset.UxDataset(xrds, uxgrid=self.uxgrid)
 
+    def to_trimesh(self,
+                   projection: Optional = None,
+                   override: Optional[bool] = False,
+                   cache: Optional[bool] = True):
+        """Constructs a ``holoviews.Trimesh`` object with triangles
+        representing the triangulation of the coordinates (nodes, edge centers,
+        or face centers) that a data variable maps to.
+
+        Parameters
+        ----------
+        projection : ccrs
+            Cartopy Projection to project coordinates to
+        override : bool
+            Flag to recompute the ``PolyCollection`` stored under the ``uxgrid`` if one is already cached
+        cache : bool
+            Flag to indicate if the computed ``PolyCollection`` stored under the ``uxgrid`` accessor should be cached
+        """
+        from holoviews import Nodes, TriMesh
+        import holoviews as hv
+        from holoviews import opts
+        import cartopy.crs as ccrs
+
+        # set extension to support opts
+        hv.extension("bokeh")
+        if self._node_centered():
+            # select node coordinates
+            element = "nodes"
+            lon, lat = self.uxgrid.node_lon.values, self.uxgrid.node_lat.values
+        elif self._edge_centered():
+            # select edge-center coordinates
+            element = "edges"
+            lon, lat = self.uxgrid.edge_lon.values, self.uxgrid.edge_lat.values
+        elif self._face_centered():
+            # select face-center coordinates
+            element = "faces"
+            lon, lat = self.uxgrid.face_lon.values, self.uxgrid.face_lat.values
+        else:
+            raise ValueError(
+                "Data is not mapped to nodes, edge centers, or face centers.")
+
+        if projection is not None:
+            # perform geographic projection
+            lon, lat, _ = projection.transform_points(ccrs.PlateCarree(), lon,
+                                                      lat).T
+
+        # obtain triangle simplices
+        simplices = self.uxgrid.to_simplices(element, projection, override,
```

Comment:
This is not taking the connectivity information in the data into account that ends up creating triangles where there is no existence of faces in the grid, which you also exemplified in the usage example section "Triangulation of a Partial-Coverage Grid" (e.g. land triangles). This is hard to be acceptable because we plot as if there is data. We need to use connectivity to overcome this (you can investigate the implementation in the feature branch [plot_rasterize](https://github.com/UXARRAY/uxarray/blob/plot_rasterize/uxarray/plot/dataarray_plot.py), e.g. `_grid_to_hvTriMesh`, `_create_hvTriMesh` that use connectivity)

Reply:
Thank you for pointing this out. I think now after you've explained it, it finally clicked with me.

Since we already know the connectivity and mesh structure, instead of running Delaunay triangulation on the set of points, we use triangulate each polygon. This should be much faster than Delaunay as well.

I'll put this PR back to a draft & migrate any point visualization related items (notebook, projection improvements) into a new PR)

Reply:
100%